tests/components/smarla/test_config_flow.py

```diff
@@ -32,11 +34,21 @@ async def test_config_flow(hass: HomeAssistant, mock_refresh_token_success) -> N
 
 
 @pytest.mark.parametrize("error", ["malformed_token", "invalid_auth"])
-async def test_form_error(
-    hass: HomeAssistant, request: pytest.FixtureRequest, error: str
-) -> None:
+async def test_form_error(hass: HomeAssistant, error: str, mock_connection) -> None:
     """Test we show user form on invalid auth."""
-    error_patch = request.getfixturevalue(f"{error}_patch")
+    match error:
+        case "malformed_token":
+            error_patch = patch(
+                "homeassistant.components.smarla.config_flow.Connection",
+                side_effect=ValueError,
+            )
+        case "invalid_auth":
+            error_patch = patch.object(
+                mock_connection,
+                "refresh_token",
+                new=AsyncMock(return_value=False),
+            )
```

Comment:
I think I might have recommended to combine these tests, but let's split them as ideally we dont have if statements (or match for that matter) in the tests